core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -79,6 +82,27 @@
         PROHIBITED_VALUE_CHAR_NAMES['\r'] = "<CR>";
     }
 
+    // Cached values for RequestHeaders
+    @Nullable
+    private HttpMethod method;
+    @Nullable
+    private Cookies cookies;
+    @Nullable
+    private List<LanguageRange> acceptLanguages;
+
+    // Cached values for ResponseHeaders
+    @Nullable
+    private HttpStatus status;
+    @Nullable
+    private Cookies setCookie;
+
+    // Cached values for HttpHeaders
+    @Nullable
+    private MediaType contentType;
+    @Nullable
+    private ContentDisposition contentDisposition;
+
+    private boolean isMutating;
```

Comment:
I'm not sure if we need to have this as a field. Could be doable with a parameter.